New package: seatd-0.4.0

ifreund opened the pull request:
~~This is not ready to be merged as upstream does not yet version the shared library files. Also, tests are currently patched out due to build failure in release mode (this is already fixed upstream but unreleased).~~

~~The only consumer of this daemon/library is the wlroots master branch as far as I know, so it doesn't really make sense to merge before the next wlroots release anyways.~~

Edit: this is good to merge

I expect that seatd will become a popular alternative to elogind on void linux, so I've made this preliminary package to facilitate early testing.

ericonr commented on srcpkgs/seatd/template:
```diff
@@ -0,0 +1,36 @@
+revision=1
+build_style=meson
+configure_args="-Dexamples=disabled -Dlogind=enabled"
+hostmakedepends="pkg-config	scdoc"
```
I think there's a tab instead of a space there.

ifreund replied:
heh, I think I may have accidentally replaced all spaces with tabs instead of just the indentation ones.

ericonr commented on srcpkgs/seatd/template:
```diff
@@ -0,0 +1,36 @@
+version=0.2.0
+revision=1
+build_style=meson
+configure_args="-Dexamples=disabled -Dlogind=enabled"
```
Make `elogind` a build option, perhaps?

ifreund replied:
Yes, it should be optional and we need a parallel option in wlroots.

ericonr replied:
Is wlroots going to move exclusively to seatd? Would be nice if we could control everything with libseat build options.

ifreund replied:
I that's the long term plan as far as I know, but I wouldn't expect that to happen before the next wlroots release. Agreed that that being able to control everything with libseat build options would be best.

ericonr commented on srcpkgs/seatd/template:
```diff
@@ -0,0 +1,36 @@
+# Template file	for	'seatd'
```
Tabs here too

ericonr commented on srcpkgs/seatd/template:
```diff
@@ -0,0 +1,36 @@
+configure_args="-Dexamples=disabled -Dlogind=enabled"
+hostmakedepends="pkg-config	scdoc"
+makedepends="elogind-devel"
+short_desc="Minimal	seat management	daemon"
```
and here

ericonr commented on srcpkgs/seatd/template:
```diff
@@ -0,0 +1,36 @@
+
+libseat-devel_package()	{
+	depends="libseat>=${version}_${revision} ${makedepends}"
+	short_desc="Universal seat management library -	development	files"
```
and here

ericonr commented on srcpkgs/seatd/files/seatd/run:
```diff
@@ -0,0 +1,2 @@
+#!/bin/sh
+exec /usr/bin/seatd -g video
```
This should probably source a conf file, right? So people can enable other `-g` options, if we get more of them.

ifreund replied:
not sure what the best option is here, and it may change as seatd matures. The `-g` flag sets the group which owns seatd's socket. The permissions of this socket determine which users are allowed access to seatd's capabilities. The `video` group is a reasonable default in my opinion, but we could also go with a seatd-specific group and ask people to add themselves to it in the install message.

ericonr replied:
Oh, I see! I would be in favor of a seatd group, I think.

ericonr commented:
I believe the build is trying to use `scdoc` from the target, not the host. Should look into how that's being done.

ifreund commented:
> I believe the build is trying to use scdoc from the target, not the host. Should look into how that's being done.

Added a fix for this to the patch, will submit it upstream in a minute as well. 

Also changed the service to pass `-g seatd` and added an install message.

ifreund commented:
This should be ready to go now. Upstream pushed a new tag including fixes for the issues mention and versioning the `.so`.

I also added a default disabled build option for elogind support as I think most users of seatd will be using it because they do not want to use elogind.

ericonr commented on common/shlibs:
```diff
@@ -3990,3 +3990,4 @@ libneatvnc.so.0 neatvnc-0.2.0_1
 libtdjson.so.1.6.0 libtd-1.6.0_1
 libJudy.so.1 judy-1.0.5_1
 libsignal-protocol-c.so.2 libsignal-protocol-c-2.3.3_2
+libseat.so.1 libseat-0.3.0_1
```
If you want to avoid rebasing all the time, throw this line near another related library (perhaps libelogind?)

ericonr commented on srcpkgs/seatd/template:
```diff
@@ -0,0 +1,39 @@
+libseat_package() {
+	short_desc="Universal seat management library"
+	pkg_install() {
+		vmove usr/lib/*.so.*
```
I believe `vmove`s are usually quoted.

ifreund replied:
certainly couldn't hurt, done.

ifreund commented:
Updated to seatd `0.4.0`

Duncaen commented on srcpkgs/seatd/INSTALL.msg:
```diff
@@ -0,0 +1,6 @@
+option which sets the group owning the socket. Thus, you will most
+likely want to add your user to the seatd group:
+
+# usermod -aG seatd <username>
```
Install messages are not documentation.

ifreund replied:
The fact that the `seatd` group in particular is used is because of the `-g seatd` flag passed in the `run` file and is specific to void's package. Is that still not worth mentioning? I'd be happy to remove the install message as well.

Duncaen replied:
Install messages spam the output and users don't read them, adding more just makes users more unlikely to read them if there are actually important messages.

You could add a README.voidlinux file like the sqmail package does.

ifreund replied:
Eh, I just removed the `INSTALL.msg`, I think this info is probably a better fit for the "Session and Seat Management" void docs page. The seatd man page and the 1 line `run` file should be plenty for most users to figure out how to use it.

ofiala-a51 commented:
Not sure if I misunderstood anything, but wouldn't it be better to separate `libseat` from `seatd` and make `seatd` depend on it? This way when `wlroots` with `libseat` support gets released (it's been implemented for some time now, so this should be soon), it could be compiled with `libseat` support instead of `elogind` and depend on `libseat`.

ifreund commented:
> Not sure if I misunderstood anything, but wouldn't it be better to separate libseat from seatd and make seatd depend on it? This way when wlroots with libseat support gets released (it's already implemented in master), it could be compiled with libseat support instead of elogind and depend on libseat.

`libseat` is separated from seatd, but seatd doesn't depend on it. `seatd` is a standalone daemon while `libseat` is a library for programs such as a wayland compositor to interact with that daemon (and optionally logind as well).

When the next wlroots version is released it will indeed depend on libseat by default, though I'm not sure we should remove `elogind` dependence by default just yet. These decisions can be exposed as build options.  of course.

ofiala-a51 commented:
I see. So shouldn't a separate PR for `libseat` be made (since this package only adds `seatd`)?

ifreund commented:
> I see. So shouldn't a separate PR for `libseat` be made (since this package only adds `seatd`)?

`libseat` is added as a subpackage in this PR

ofiala-a51 commented on srcpkgs/seatd/template:
```diff
@@ -0,0 +1,39 @@
+homepage="https://kl.wtf/projects/seatd/"
+distfiles="https://git.sr.ht/~kennylevinsen/seatd/archive/${version}.tar.gz"
+checksum=3c7458ff0c0d41ddd7734d01a5c5a9110ee5b94f01931e7e41ea0fd0c2e6050f
+system_groups=seatd
```
The group should be prefixed by an underscore (`_seatd`) as [described in Manual.md](https://github.com/void-linux/void-packages/blob/master/Manual.md#creating-system-accountsgroups-at-runtimel)

ifreund replied:
That applies only to accounts not groups, assuming I read the manual correctly.

Duncaen replied:
Generally applies to both as a type of namespace between system generated groups besides the base set of groups and user created groups.

ifreund replied:
Seems reasonable, we should make the wording in the manual more clear

ifreund replied:
#26230

ofiala-a51 commented on srcpkgs/seatd/files/seatd/run:
```diff
@@ -0,0 +1,2 @@
+#!/bin/sh
+exec /usr/bin/seatd -g seatd
```
`_seatd`

ofiala-a51 commented on srcpkgs/seatd/template:
```diff
@@ -0,0 +1,39 @@
+version=0.4.0
+revision=1
+build_style=meson
+configure_args="-Dexamples=disabled $(vopt_if elogind -Dlogind=enabled)"
```
elogind support should be compiled in by default to support switch to build wlroots w/ libseat backend only; see #26224

PaperMountainStudio replied:
it should be enabled by default:
> build_options_default="elogind"

ericonr commented:
This is ready to be merged, right?

ifreund commented:
> This is ready to be merged, right?

@ericonr Indeed, this PR is good to go.
